packages/authorization-server/src/model/domain/errors.ts

```diff
@@ -0,0 +1,138 @@
+      request
+    )}`,
+    code: "clientAssertionRequestValidationFailed",
+    title: "Client assertion request validation failed",
+  });
+}
+
+export function clientAssertionValidationFailed(
+  clientAssertion: string,
+  clientId: string | undefined
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Client assertion validation failed for clientAssertion: ${clientAssertion}, clientId: ${clientId}`,
+    code: "clientAssertionValidationFailed",
+    title: "Client assertion validation failed",
+  });
+}
+
+export function clientAssertionSignatureValidationFailed(
+  clientAssertion: string
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Client assertion signature validation failed for clientAssertion: ${clientAssertion}`,
+    code: "clientAssertionSignatureValidationFailed",
+    title: "Client assertion signature validation failed",
+  });
+}
+
+export function kafkaAuditingFailed(): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: "Kafka auditing failed ",
+    code: "kafkaAuditingFailed",
+    title: "Kafka auditing failed",
+  });
+}
+
+export function fallbackAuditFailed(jti: string): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Fallback audit failed. Jti: ${jti}`,
+    code: "fallbackAuditFailed",
+    title: "Fallback audit failed",
+  });
+}
+
+export function tokenGenerationStatesEntryNotFound(
+  pk: TokenGenerationStatesClientKidPurposePK | TokenGenerationStatesClientKidPK
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Entry with PK ${pk} not found in token-generation-states table`,
+    code: "tokenGenerationStatesEntryNotFound",
+    title: "token-generation-states entry not found",
+  });
+}
+
+export function keyRetrievalFailed(): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: "Key retrieval failed",
+    code: "keyRetrievalFailed",
+    title: "Key retrieval failed",
+  });
+}
+
+export function invalidTokenClientKidPurposeEntry(): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail:
+      "Missing data in client-kid-purpose entry from token-generation-states table",
+    code: "invalidTokenClientKidPurposeEntry",
+    title: "Invalid token client-kid-purpose entry",
+  });
+}
+
+export function keyTypeMismatch(
+  prefix: string,
+  clientKind: ClientKindTokenStates
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Token-generation entry with prefix ${prefix} can't have client kind: ${clientKind}`,
+    code: "keyTypeMismatch",
+    title: "Key type mismatch",
+  });
+}
+
+export function unexpectedTokenGenerationStatesEntry(): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: "Unexpected token-generation-states entry",
+    code: "unexpectedTokenGenerationStatesEntry",
+    title: "Unexpected token-generation-states entry",
+  });
+}
+
+export function unexpectedTokenGenerationError(): ApiError<ErrorCodes> {
```

Comment:
this is used in a point where we can better identify the error.
Be sure that the name of the error and the description give enough data.
Plus, let's add the `clientId`

Reply:
Removed after a refactor